### PR TITLE
improve buildOpAssign()

### PR DESCRIPTION
**Pull request by WalterBright**
Should make it easier to read.

### CODE REVIEW

File: src/dmd/expression.d [hunk 1]
```diff
@@ -1722,6 +1722,16 @@ extern (C++) abstract class Expression : RootObject
         return e1;
     }
 
+    static Expression combine(Expression e1, Expression e2, Expression e3)
+    {
+        return combine(combine(e1, e2), e3);
+    }
+
+    static Expression combine(Expression e1, Expression e2, Expression e3, Expression e4)
```
Comment (JinShil) [inline src/dmd/expression.d]: These two functions seem a little weird.  Can we make it a single variadic; maybe a template?

File: src/dmd/clone.d [hunk 2]
```diff
@@ -141,20 +145,32 @@ extern (C++) FuncDeclaration hasIdentityOpAssign(AggregateDeclaration ad, Scope*
 private bool needOpAssign(StructDeclaration sd)
 {
     //printf("StructDeclaration::needOpAssign() %s\n", sd.toChars());
-    if (sd.isUnionDeclaration())
+
+    static bool need()
```
Comment (JinShil) [inline src/dmd/clone.d]: Naming conventions, as far as I'm aware, would prefer `isNeeded()` and `isNotNeeded`.

Comment (jacob-carlborg) [reply src/dmd/clone.d]: Do we really need both?

File: src/dmd/expression.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/expression.d]: It's obvious what they do with a quick glance. Using variadics or templates, not so easy.

**Review by RazvanN7 — APPROVED** [review]
I'll merge this and make a new PR which addresses the reviewers comments

**Pull request merged**